crates/sui-package-resolver/src/error.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::VMError;
+use move_core_types::account_address::AccountAddress;
+use sui_indexer::errors::IndexerError as DbError;
+use thiserror::Error;
+use typed_store::TypedStoreError;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("{0}")]
+    Bcs(#[from] bcs::Error),
+
+    #[error("{0}")]
+    Db(#[from] DbError),
+
+    #[error("{0}")]
+    TypedStore(#[from] TypedStoreError),
```

Comment:
Now that we have multiple stores under the hood, we will likely need to make this error type generic, with an underlying `StoreError`:

```rust
pub enum Error<StoreError: Error> {
    #[error(transparent)]
    Store(#[from] StoreError),
}
```

Reply:
The better approach would likely be to use a trait object here:
```
pub enum Error {
    Store(Box<dyn Error + Send + 'static>),
}
```

imo theres little value in having generic error types and it makes things harder to work with. If you do need to recover the concrete type you can leverage downcasting

Reply:
That's fine as well.